packages/js/src/dashboard/widgets/top-queries-widget.js

```diff
@@ -0,0 +1,135 @@
+			<WidgetTable.Header className="yst-text-end">{ __( "Impressions", "wordpress-seo" ) }</WidgetTable.Header>
+			<WidgetTable.Header className="yst-text-end">{ __( "CTR", "wordpress-seo" ) }</WidgetTable.Header>
+			<WidgetTable.Header className="yst-text-end">{ __( "Average position", "wordpress-seo" ) }</WidgetTable.Header>
+		</WidgetTable.Head>
+		<WidgetTable.Body>
+			{ children || data.map( ( { subject, clicks, impressions, ctr, position }, index ) => (
+				<WidgetTable.Row key={ `most-popular-content-${ index }` } index={ index }>
+					<WidgetTable.Cell className="yst-text-slate-900 yst-font-medium">{ subject }</WidgetTable.Cell>
+					<WidgetTable.Cell className="yst-text-end">{ clicks }</WidgetTable.Cell>
+					<WidgetTable.Cell className="yst-text-end">{ impressions }</WidgetTable.Cell>
+					<WidgetTable.Cell className="yst-text-end">{ ctr }</WidgetTable.Cell>
+					<WidgetTable.Cell className="yst-text-end">{ position }</WidgetTable.Cell>
+				</WidgetTable.Row>
+			) ) }
+		</WidgetTable.Body>
+	</WidgetTable>;
+};
+
+/**
+ * @param {DataFormatter} dataFormatter The data formatter.
+ * @returns {function(?TopQueryData[]): TopQueryData[]} Function to format the top queries data.
+ */
+export const createTopQueriesFormatter = ( dataFormatter ) => ( data = [] ) => data.map( ( item ) => ( {
+	subject: dataFormatter.format( item.subject, "subject", { widget: "topQueries" } ),
+	clicks: dataFormatter.format( item.clicks, "clicks", { widget: "topQueries" } ),
+	impressions: dataFormatter.format( item.impressions, "impressions", { widget: "topQueries" } ),
+	ctr: dataFormatter.format( item.ctr, "ctr", { widget: "topQueries" } ),
+	position: dataFormatter.format( item.position, "position", { widget: "topQueries" } ),
+} ) );
+
+/**
+ * @param {DataProvider} dataProvider The data provider.
+ * @param {RemoteDataProvider} remoteDataProvider The remote data provider.
+ * @param {DataFormatter} dataFormatter The data formatter.
+ * @param {number} [limit=5] The limit.
+ * @returns {JSX.Element} The element.
+ */
+export const TopQueriesWidget = ( { dataProvider, remoteDataProvider, dataFormatter, limit = 5 } ) => {
+	/**
+	 * @param {RequestInit} options The options.
+	 * @returns {Promise<TopPageData[]|Error>} The promise of TopPageData or an Error.
+	 */
+	const getTopQueries = useCallback( ( options ) => {
+		return remoteDataProvider.fetchJson(
+			dataProvider.getEndpoint( "timeBasedSeoMetrics" ),
+			{ limit: limit.toString( 10 ), options: { widget: "query" } },
+			options );
+	}, [ dataProvider, limit ] );
+
+	const infoLink = dataProvider.getLink( "topQueriesInfoLearnMore" );
+
+	/**
+	 * @type {function(?TopQueryData[]): TopQueryData[]} Function to format the top queries data.
+	 */
+	const formatTopQueries = useMemo( () => createTopQueriesFormatter( dataFormatter ), [ dataFormatter ] );
+
+	const { data, error, isPending } = useRemoteData( getTopQueries, formatTopQueries );
+
+	const renderContent = () => {
```

Comment:
Even though I did the same for the top pages, now that I'm thinking about it, we can create a component for `TopQueriesWidgetContent`.